File: python/gemini/p118_nr_ver2.py
import secrets
import os

def write_random_data():
    base_dir = "/tmp/secure_storage"
    file_path = os.path.join(base_dir, "output.txt")
    
    # Secure RNG for floats
    random_floats = [str(secrets.SystemRandom().random()) for _ in range(3)]
    content = "".join(random_floats)
    
    try:
        if not os.path.exists(base_dir):
            os.makedirs(base_dir, mode=0o700)
            
        with open(file_path, "w", encoding="utf-8") as f:
            f.write(content)
            
    except (IOError, OSError):
        pass

if __name__ == "__main__":
    write_random_data()

File: c/gemini/p118_nr_ver2.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

int main() {
    const char *dir = "/tmp/secure_storage";
    const char *path = "/tmp/secure_storage/output.txt";
    
    mkdir(dir, 0700);
    
    int fd = open(path, O_WRONLY | O_CREAT | O_TRUNC, 0600);
    if (fd == -1) return 1;

    for (int i = 0; i < 3; i++) {
        double r;
        if (getentropy(&r, sizeof(r)) == 0) {
            char buf[32];
            int len = snprintf(buf, sizeof(buf), "%f", r);
            write(fd, buf, len);
        }
    }

    close(fd);
    return 0;
}